server/models/message.js

```diff
@@ -11,10 +11,18 @@ module.exports = (DataTypes, sequelize) => {
             type: DataTypes.INTEGER,
             allowNull: false,
         },
+        senderName: {
+            type: DataTypes.STRING,
+            allowNull: false,
+        },
         receiverId: {
             type: DataTypes.INTEGER,
             allowNull: false,
         },
+        receiverName: {
+            type: DataTypes.STRING,
+            allowNull: false,
+        },
```

Comment:
These fields are duplicates of the `username` field in `abstractUser` model. If you need to access these fields, do a join on the `abstractUser` model. Please get rid of them here

Reply:
I can't speak to recieverName @QinAlex can elaborate I am sure, but senderName has to be stored with the message else we lose access to it if that user is deleted. I.e Receiver keeps the message, but it looks like it came from from no one. @rachellegelden